scripts/generate_sitemap.py

```diff
@@ -0,0 +1,195 @@
+        tag_text = self.doc.createTextNode(text)
+        tag.appendChild(tag_text)
+
+    def add_url(self, **kwargs):
+        """Adds a url to the current urlset"""
+        if self.url_count > self.url_max:
+            self.write_doc()
+            self.new_doc()
+        self.url = self.doc.createElement('url')
+        self.urlset.appendChild(self.url)
+
+        for k, v in kwargs.iteritems():
+            self.add_tag(k, v)
+        self.url_count += 1
+
+    def write_doc(self):
+        """Writes and gzips each sitemap xml file"""
+        path = os.path.join(self.sitemap_dir, 'sitemap_{}.xml'.format(str(self.sitemap_count)))
+        print('Writing and gzipping `{}`: url_count = {}'.format(path, str(self.url_count)))
+
+        xml_str = self.doc.toprettyxml(indent="  ", encoding='utf-8')
+        with open(path, 'w') as f:
+            f.write(xml_str)
+        # Write zipped file
+        with open(path, 'rb') as f_in, gzip.open(path + '.gz', 'wb') as f_out:
+            shutil.copyfileobj(f_in, f_out)
+        self.sitemap_count += 1
+
+    def write_sitemap_index(self):
+        """Writes the index file for all of the sitemap files"""
+        doc = xml.dom.minidom.Document()
+        sitemap_index = self.doc.createElement('sitemapindex')
+        doc.appendChild(sitemap_index)
+
+        for f in range(self.sitemap_count):
+            sitemap = doc.createElement('sitemap')
+            sitemap_index.appendChild(sitemap)
+            
+            loc = doc.createElement('loc')
+            sitemap.appendChild(loc)
+            loc_text = self.doc.createTextNode(urlparse.urljoin(settings.DOMAIN, 'sitemaps/sitemap_{}.xml.gz'.format(str(f))))
+            loc.appendChild(loc_text)
+
+            datemod = doc.createElement('datemod')
+            sitemap.appendChild(datemod)
+            datemod_text = self.doc.createTextNode(datetime.datetime.now().isoformat())
+            datemod.appendChild(datemod_text)
+
+        print('Writing `sitemap_index.xml`')
+        xml_str = doc.toprettyxml(indent="  ", encoding='utf-8')
+        with open(os.path.join(self.sitemap_dir, 'sitemap_index.xml'), 'w') as f:
+            f.write(xml_str)
+
+    def generate(self):
+
+        # Static urls
+        static_urls = [
+            ['', 'yearly'],
+            ['preprints', 'yearly'],
+            ['prereg', 'yearly'],
+            ['meetings', 'yearly'],
+            ['registries', 'yearly'],
+            ['explore/activity', 'weekly'],
+            ['support', 'yearly'],
+            ['faq', 'yearly'],
+        ]
+
+        for url, freq in static_urls:
+            self.add_url(**{
+                'loc': urlparse.urljoin(settings.DOMAIN, url),
+                'changefreq': freq,
+            })
+
+        # User urls
+        for obj in OSFUser.objects.filter(is_active=True):
+            self.add_url(**{
+                'loc': urlparse.urljoin(settings.DOMAIN, obj.url),
+                'changefreq': 'yearly',
```

Comment:
Depending on how fancy we get with this, we'd likely want to make this the configurable part.
```
'changefreq`: settings.SITEMAP_FREQUENCY.OSF_USER
```
Or similar.